src/fcntl.rs

```diff
@@ -629,6 +632,102 @@ pub fn flock(fd: RawFd, arg: FlockArg) -> Result<()> {
 
     Errno::result(res).map(drop)
 }
+
+/// Represents valid types for flock.
+///
+/// # Safety
+/// `T` must be `!Clone`.
```

Comment:
Nit: might be better to rephrase this:

> Types implementing this must be `!Clone`

cause there is no `T` defined in this trait